packages/core/src/utils/vercel-ai.ts

```diff
@@ -60,13 +60,34 @@ function onVercelAiSpanStart(span: Span): void {
   processGenerateSpan(span, name, attributes);
 }
 
+interface TokenSummary {
+  inputTokens: number;
+  outputTokens: number;
+}
+
 function vercelAiEventProcessor(event: Event): Event {
   if (event.type === 'transaction' && event.spans) {
+    // Map to accumulate token data by parent span ID
+    const tokenAccumulator: Map<string, TokenSummary> = new Map();
+
+    // First pass: process all spans and accumulate token data
     for (const span of event.spans) {
-      // this mutates spans in-place
       processEndedVercelAiSpan(span);
+
+      // Accumulate token data for parent spans
+      accumulateTokensForParent(span, tokenAccumulator);
+    }
+
+    // Second pass: apply accumulated token data to parent spans
+    for (const span of event.spans) {
+      if (span.op !== 'gen_ai.invoke_agent') {
```

Comment:
```suggestion
      if (span.op !== 'gen_ai.invoke_agent' || !tokenAccumulator.has(span.span_id)) {
```

should also be a reasonable shortcut?